src/test/java/com/connexta/store/adaptors/S3StorageAdaptorITests.java

```diff
@@ -153,11 +157,42 @@ public void testStoreWithContentLengthNotMatching() {
               10L,
               MediaType.APPLICATION_XML_VALUE,
               new ByteArrayInputStream(ASDF.getBytes()),
-              KEY,
+              DATASET_ID,
               Map.of());
         });
   }
 
+  @Test
+  void testUpdatingStoreStatus() {
+    // Store a file and set status as staged
+    storageAdaptor.store(
+        4L,
+        MediaType.APPLICATION_XML_VALUE,
+        new ByteArrayInputStream(ASDF.getBytes()),
+        DATASET_ID,
+        Map.of());
+    assertEquals(STAGED, getStatusTag(DATASET_ID));
```

Comment:
✏️ Unit test methods should only test one thing. I suggest moving this assert to `testSuccessfulStoreRequest` so that this method only asserts the new status.